protocol/contracts/nexus/Shards.sol

```diff
@@ -0,0 +1,232 @@
+//                                                .@@@@@@@@@@@@@@@                
+//                                                .@@           @@                
+//                                                .@@           @@                
+//                       @@@&&&&&&&&&&&&&&&&&&&&&&&@@&&&&@@(    @@                
+//                       @@& @@@@@@               .@@    @@(    @@                
+//                       @@& @@@@@@               .@@@@@@@@@@@@@@@                
+//                       @@&                             @@(                      
+//                       @@&                             @@(                      
+//                       @@&                             @@(                      
+//                       @@&                             @@(                      
+//                       @@&                             @@(                      
+//                       @@&                             @@(                      
+//                       @@&                         @@@ @@(                      
+//                 @@@@@@@@@@@@@@@                   @@@ @@(                      
+//                 @@    @@&    @@                @@@@@@ @@(                      
+//                 @@    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@(                      
+//                 @@           @@                                                
+//                 @@           @@                                                
+//                 @@@@@@@@@@@@@@@   
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+
+interface IRelic { 
+    function getRelicId(address _owner) external view returns (uint256);
+    function hasRelic(address _owner) external view returns (bool);
+    function ownerOf(uint256 tokenId) external view returns (address);
+    function balanceOf(address _owner) external view returns(uint256);
+}
+
+
+contract Shards is
+    ERC1155,
+    Ownable,
+    Pausable,
+    ERC1155Burnable,
+    ERC1155Supply,
+    ReentrancyGuard
+{
+    constructor() ERC1155("") {}
+
+    // whitelisted partners to mint Shards for users
+    mapping (address => bool) public partnerMinters;
```

Comment:
Do you need this `partnerMinters` mapping when you have the `partnerAllowedIds` mapping already?

If the partner is not in `partnerAllowedIds` then it won't have any ids that it's allowed to mint anyway